tooling/minimize/tests/ub/enum_mark_used_bytes.rs

```diff
@@ -0,0 +1,35 @@
+
+/// This test case ensures that enums correctly mark their used bytes for unions and don't mark padding.
+/// It creates a union with all initialized bytes, and then after a copy tries to read all bytes.
+/// The read should fail as padding will get uninitialized during the copy.
+/// For alignment reasons the bytes are read as `u16`, as the enum is 2-byte aligned.
+
+#[derive(Clone, Copy)]
+#[repr(u16)]
+#[allow(unused)]
+enum Inner {
+    WithData(u16, u8),
+    Empty,
+}
+
+#[derive(Clone, Copy)]
+union TestUnion {
+    _data: Inner
+}
+
+// Shorthand for an array of `u16` with the same size of the union.
+type UnionShorts = [u16; std::mem::size_of::<TestUnion>() / 2];
+
+fn get_union_shorts(u: TestUnion) -> UnionShorts {
+    unsafe { *(&u as *const TestUnion as *const UnionShorts) }
+}
+
+
+fn main() {
+    let u = unsafe {
+        let zero = [0u16; std::mem::size_of::<TestUnion>() / 2];
+        *(&zero as *const UnionShorts as *const TestUnion)
+    };
```

Comment:
I think this is easier when expressed as a transmute. Then you also don't have to worry about alignment.
```rust
let zero = [0u8; std::mem::size_of::<TestUnion>()];
let u: TestUnion = transmute(zero);
// This is UB because the padding bytes get de-initialized.
let _ints: [0u8; std::mem::size_of::<TestUnion>()] = transmute(u);
```

Reply:
good point